Random fuzz testing (#608)
foundry-rs/starknet-foundry: docs/src/testing/advanced/fuzz-testing.md

# Fuzz Testing

In many cases, a test needs to verify function behavior for multiple possible values.
While it is possible to come up with these cases on your own, it is often impractical, especially when you want to test

> for a large number of possible arguments.

Reviewer: not sure if `test for` or `test against` is correct

Author: I think the second one sounds, better. Changing

> ℹ️**Info**
> Currently, Forge only supports using randomly generated values for fuzzing.
> This way of fuzzing doesn't support any kind of value generation based on code analysis, test coverage or results of
> other fuzzer runs.

> > In the future, more advanced fuzzing execution modes will be added.

Author: I don't think this note makes a lot of sense if you remove `for fuzzing`.

## Random Fuzzing

> ⚠️ **Warning**

> > Currently Forge only supports fuzz testing arguments with `felt252` type. Trying to use different argument types will

Reviewer 2: There was a section in the docs about the type of test functions. It should be adjusted in this PR

> result in an error.

To convert a test to a random fuzz test, simply add arguments to the test function.
These arguments can then be used in the test body.
The test will be run multiple times for different randomly generated values.

```cairo
fn sum(a: felt252, b: felt252) -> felt252 {
    return a + b;
}

#[test]
fn test_sum(x: felt252, y: felt252) {
    assert(sum(x, y) == x + y, 'sum incorrect');
}
```

Then run `snforge` like usual.

```shell
$ snforge
Collected 1 test(s) and 1 test file(s)
Running 1 test(s) from package_name package
Running fuzzer for package_name::test_sum, 256 runs:
[PASS] package_name::test_sum
```

## Configuring The Fuzzer

> It is possible to configure the number of runs of the random fuzzer as well as its seed too by a command line argument:

Reviewer: ```suggestion
It is possible to configure the number of runs of the random fuzzer as well as its seed too with command line arguments:
```
